core/src/main/java/edu/wpi/grip/core/sources/MultiImageFileSource.java

```diff
@@ -0,0 +1,139 @@
+        this.initialize(eventBus, files.stream()
+                .map(file -> URLDecoder.decode(Paths.get(file.toURI()).toString()))
+                .collect(Collectors.toList()), index);
+    }
+
+    public MultiImageFileSource(final EventBus eventBus, final List<File> files) throws IOException {
+        this(eventBus, files, 0);
+    }
+
+    /**
+     * Used only for serialization
+     */
+    public MultiImageFileSource(){ /* no op */ }
+
+
+    @SuppressWarnings("unchecked")
+    private void initialize(final EventBus eventBus, final List<String> paths, int index) throws IOException {
+        this.eventBus = checkNotNull(eventBus, "Event bus can not be null");
+        this.index = checkElementIndex(index, paths.size(), "File List Index");
+        this.paths = paths;
+        this.outputSocket = new OutputSocket(eventBus, imageOutputHint);
+        for (String path : paths) {
+            // Ensure that all of the images can be loaded
+            loadImage(path);
+        }
+        loadImage(paths.get(index));
+    }
+
+    @Override
+    public String getName() {
+        return "Multi-Image";
+    }
+
+    @Override
+    protected OutputSocket[] createOutputSockets() {
+        return new OutputSocket[]{
+                outputSocket
+        };
+    }
+
+    @Override
+    public Properties getProperties() {
+        final Properties properties = new Properties();
+        properties.setProperty(SIZE_PROPERTY, Integer.toString(paths.size()));
+        properties.setProperty(INDEX_PROPERTY, Integer.toString(index));
+        for (int i = 0; i < paths.size(); i++) {
+            properties.setProperty(getPathProperty(i), paths.get(i));
+        }
+        return properties;
+    }
+
+    @Override
+    public void createFromProperties(EventBus eventBus, Properties properties) throws IOException {
+        final int index = Integer.valueOf(properties.getProperty(INDEX_PROPERTY));
+        final int size = Integer.valueOf(properties.getProperty(SIZE_PROPERTY));
+        final List<String> paths = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            paths.add(properties.getProperty(getPathProperty(i)));
+        }
+        this.initialize(eventBus, paths, index);
+    }
+
+
+    private void loadImage(String path) throws IOException {
+        OpenCVUtility.loadImage(path, this.outputSocket.getValue().get());
+        this.outputSocket.setValue(this.outputSocket.getValue().get());
+        this.eventBus.post(new SourceStartedEvent(this));
+    }
+
+
+    private int getListIndex(){
+        final int listSize = paths.size();
+        int indexToGet = index % listSize;
+        //this might happen to be negative
+        return indexToGet < 0 ? indexToGet + listSize : indexToGet;
+    }
+
+    @Override
+    public synchronized void nextValue() throws IOException {
+        index ++;
+        loadImage(paths.get(getListIndex()));
```

Comment:
You should not load images again and again when switching - just load all of them once when you create the source.  This seems like it could be a real problem if you have somewhat high quality images and you don't have an SSD